src/main/java/frc/team2412/robot/subsystems/ArmSubsystem.java

```diff
@@ -88,75 +146,168 @@ public static class ArmConstants { // To find values later
+				MathUtil.clamp(MAX_ARM_VELOCITY * percentOutput, MIN_PERCENT_OUTPUT, MAX_PERCENT_OUTPUT));
+	}
 
-	// public double getHorizontalArmPos() {
-	// 	return ((ARM_LENGTH * Math.cos(getShoulderAngle()))
-	// 			+ (VIRTUAL_BAR_LENGTH * Math.cos(getElbowAngle())));
-	// }
+	public void setWristMotor(double percentOutput) {
+		wristMotor.set(
+				MathUtil.clamp(MAX_WRIST_VELOCITY + percentOutput, MIN_PERCENT_OUTPUT, MAX_PERCENT_OUTPUT));
+	}
 
-	/** Stops Arm */
-	public void stopArm() {
-		armMotor.stopMotor();
+	public void setPosition(PositionType position) {
+		currentPosition = position;
 	}
 
-	public void stopWrist() {
-		wristMotor.stopMotor();
+	public void setManualOverride(boolean override) {
+		manualOverride = override;
 	}
 
-	public void rotateArmTo(double targetAngle) {
-		if (getShoulderAngle() != targetAngle) {
-			armPID.setGoal(targetAngle);
-			armMotor.set(armPID.calculate(getShoulderAngle(), targetAngle));
-		}
+	public void setArmGoal(double targetAngle) {
+		armPID.setGoal(MathUtil.clamp(targetAngle, MIN_ARM_ANGLE, MAX_ARM_ANGLE));
 	}
 
-	public void rotateWristTo(double targetAngle) {
-		if (getWristAngle() != targetAngle) {
-			wristPID.setGoal(targetAngle);
-			wristMotor.set(wristPID.calculate(getWristAngle(), targetAngle));
-		}
+	public void setWristGoal(double targetAngle) {
+		wristPID.setGoal(MathUtil.clamp(targetAngle, MIN_WRIST_ANGLE, MAX_WRIST_ANGLE));
 	}
 
-	public boolean armIsAtGoal() {
-		return armPID.atGoal();
+	public double calculateArmPID() {
+		return armPID.calculate(shoulderEncoder.getDistance(), armPID.getGoal());
 	}
 
-	public boolean wristIsAtGoal() {
-		return wristPID.atGoal();
+	public double calculateWristPID() {
+		return armPID.calculate(wristEncoder.getDistance(), wristPID.getGoal());
+	}
+
+	public double calculateArmFeedforward() {
+		return ARM_K_S * Math.signum(armPID.getGoal().position - getShoulderAngle())
+				+ ARM_K_G * getAngleTowardsCenterOfMass()
+				+ ARM_K_V * 0
+				+ ARM_K_A * 0;
+	}
+
+	public double calculateWristFeedforward() {
+		return wristFeedforward.calculate(
+				shoulderEncoder.getDistance() + wristEncoder.getDistance(), 0); // getRate = getVelocity?
 	}
 
 	public double getShoulderAngle() {
 		return shoulderEncoder.getDistance();
 	}
 
+	public double getElbowAngle() {
+		return shoulderEncoder.getDistance() * SHOULDER_ELBOW_GEAR_RATIO;
+	}
+
 	public double getWristAngle() {
 		return wristEncoder.getDistance();
 	}
+
+	public PositionType getPosition() {
+		return currentPosition;
+	}
+
+	public double convertToVolts(double percentOutput) {
+		return percentOutput * Robot.getInstance().getVoltage();
+	}
+
+	public double getAngleTowardsCenterOfMass() {
+
+		// Get Coordinates of Centers of Mass
+
+		// intake = 5.57 lb
+		// angle offset for center of mass intake =
+		double innerArmCenterOfMassX = 11.218 * Math.cos(Math.toRadians(180 - getShoulderAngle()));
+		double innerArmCenterOfMassY = 11.218 * Math.sin(Math.toRadians(180 - getShoulderAngle()));
```

Comment:
Can we use the constants defined in hardware (INNER_ARM_CENTER_OF_MASS_DISTANCE_FROM_JOINT) instead of hardcoding?